media-gfx/pqiv/pqiv-2.12.ebuild

```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit edo linux-info toolchain-funcs xdg-utils
+
+SRC_URI="https://github.com/phillipberndt/pqiv/archive/${PV}.tar.gz -> ${P}.tar.gz"
+KEYWORDS="~amd64 ~arm ~arm64 ~riscv ~x86"
+
+DESCRIPTION="A powerful GTK 3 based command-line image viewer with a minimal UI"
+HOMEPAGE="https://github.com/phillipberndt/pqiv http://www.pberndt.com/Programme/Linux/pqiv/"
+
+LICENSE="GPL-3+"
+SLOT="0"
+IUSE="archive ffmpeg imagemagick pdf postscript webp"
+
+RDEPEND="
+	>=dev-libs/glib-2.32:2
+	>=x11-libs/cairo-1.6
+	>=x11-libs/gdk-pixbuf-2.2:2
+	x11-libs/gtk+:3
+	>=x11-libs/pango-1.10
+	archive? ( app-arch/libarchive:0= )
+	ffmpeg? ( media-video/ffmpeg:0= )
+	imagemagick? ( media-gfx/imagemagick:0= )
+	pdf? ( app-text/poppler:0= )
+	postscript? ( app-text/libspectre:0= )
+	webp? ( media-libs/libwebp:0= )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+pkg_setup() {
+	if use kernel_linux; then
+		CONFIG_CHECK="~INOTIFY_USER"
+		linux-info_pkg_setup
+	fi
+}
+
+src_configure() {
+	local backends=(
+		"gdkpixbuf"
+		$(usex archive "archive" "")
+		$(usex archive "archive_cbx" "")
+		$(usex ffmpeg "libav" "")
+		$(usex imagemagick "wand" "")
+		$(usex pdf "poppler" "")
+		$(usex postscript "spectre" "")
+		$(usex webp "webp" "")
+	)
+	local myconf=(
+		--backends-build=shared
+		--backends=$(printf "%s," "${backends[@]}")
+		--prefix="${EPREFIX}/usr"
+		--libdir="${EPREFIX}/usr/$(get_libdir)"
+	)
+	edo ./configure "${myconf[@]}"
+}
+
+src_compile() {
+	tc-export CC
+	emake VERBOSE=1 CFLAGS="${CFLAGS}"
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+}
+
+pkg_postrm() {
+	xdg_desktop_database_update
+}
```

Comment:
If you `inherit xdg` instead of `xdg-utils`, you can just drop those declarations (this eclass adds those phases and scans the install dir for any such file, and call appropriate update function automatically).